modules/core/src/main/java/org/apache/ignite/internal/IgniteKernal.java

```diff
@@ -1190,6 +1190,10 @@ public void start(
 
             ctx.add(discoMgr, false);
 
+            // Start the encryption manager after assigning the discovery manager to context, so it will be
```

Comment:
We should change the start order because of `EncryptionManager` should listen to custom discovery events. See `GridEncryptionManager.start()` and constructor of `DistributedProcess`.